test/uk/gov/hmrc/cgtpropertydisposals/actors/TaxEnrolmentRetryManagerSpec.scala

```diff
@@ -0,0 +1,208 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cgtpropertydisposals.actors
+
+import java.time.LocalDateTime
+import org.scalacheck.ScalacheckShapeless._
+import akka.actor.{ActorRefFactory, ActorSystem, PoisonPill, Props}
+import akka.testkit.{ImplicitSender, TestKit, TestProbe}
+import cats.data.EitherT
+import com.miguno.akka.testing.VirtualTime
+import org.scalacheck.{Arbitrary, Gen}
+import org.scalamock.scalatest.MockFactory
+import org.scalatest.{BeforeAndAfterAll, Matchers, WordSpecLike}
+import uk.gov.hmrc.cgtpropertydisposals.actors.FakeChildMaker.NewChildCreated
+import uk.gov.hmrc.cgtpropertydisposals.actors.TaxEnrolmentRetryConnectorActor.RetryCallToTaxEnrolmentService
+import uk.gov.hmrc.cgtpropertydisposals.actors.TaxEnrolmentRetryManager.{MarkTaxEnrolmentRetryRequestAsSuccess, RetryTaxEnrolmentRequest}
+import uk.gov.hmrc.cgtpropertydisposals.models.Address.UkAddress
+import uk.gov.hmrc.cgtpropertydisposals.models.{Address, Error, TaxEnrolmentRequest, sample}
+import uk.gov.hmrc.cgtpropertydisposals.repositories.TaxEnrolmentRetryRepository
+import uk.gov.hmrc.cgtpropertydisposals.service.TaxEnrolmentService
+
+import scala.concurrent.Future
+
+class TaxEnrolmentRetryManagerSpec
+    extends TestKit(ActorSystem())
+    with ImplicitSender
+    with WordSpecLike
+    with Matchers
+    with BeforeAndAfterAll
+    with MockFactory {
+
+  override def afterAll: Unit =
+    TestKit.shutdownActorSystem(system)
+
+  val time: VirtualTime = new VirtualTime()
+
+  val mockTaxEnrolmentService    = mock[TaxEnrolmentService]
+  val mockTaxEnrolmentRepository = mock[TaxEnrolmentRetryRepository]
+
+  implicit val localDateArb: Arbitrary[LocalDateTime] = Arbitrary.apply[LocalDateTime](Gen.const(LocalDateTime.now()))
+  val taxEnrolmentRequest                             = sample[TaxEnrolmentRequest]
+
+  "A Tax Enrolment Retry Manager actor" must {
+
+    "send a message to the mongo actor if tax enrolnment call was successful" in {
+      val mongoProbe     = TestProbe()
+      val connectorProbe = TestProbe()
+
+      val app = system.actorOf(
+        Props(
+          new TaxEnrolmentRetryManager(
+            (_: ActorRefFactory) => new FakeChildMaker(self, mongoProbe.ref).newChild(),
+            (_: ActorRefFactory) => new FakeChildMaker(self, connectorProbe.ref).newChild(),
+            mockTaxEnrolmentRepository,
+            mockTaxEnrolmentService,
+            system.scheduler
```

Comment:
you need to pass in time.scheduler here to make the advancing of time be picked up

Reply:
I think I did try that and it didn't work as well. I will give it another go in case I missed something.

Reply:
I switched back to time.scheduler from the VirtualTime class and advanced the schedule but it doesn't work. I expect the connectorProbe to receive one message when the time has been advanced to represent the re-scheduling of the failed tax enrolment call but that doesn't seem to happen. I am not sure if the scheduler is being notified of the advancement in time.